### PR TITLE
Module to create resource group

### PR DESCRIPTION
**Pull request by umarali-nagoor**
Screenshot 2021-05-10 at 2 52 05 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Full examples are in the [examples](./examples/) folder, but basic usage is as follows for creation of namespace and action :
+
+```hcl
+provider "ibm" {
+}
+
+module "resource_groups" {
+  source = "terraform-ibm-modules/cluster/ibm//modules/resource-group"
+  resource_group_name = var.resource_group_name
```
Comment (hkantare) [inline README.md]: resource_group_name this shd be name

File: examples/create-resource-group/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+This module is used to create a resource group.
+
+## Example Usage
+```
+
+module "resource_groups" {
+  source = "terraform-ibm-modules/cluster/ibm//modules/resource-group"
+
+
+  resource_group_name = var.name
```
Comment (hkantare) [inline examples/create-resource-group/README.md]: resource_group_name this shd be name

File: modules/resource-group/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+This module is used to create a resource group.
+
+## Example Usage
+```
+
+module "resource_groups" {
+  source = "terraform-ibm-modules/cluster/ibm//modules/resource-group"
+
+
+  resource_group_name = var.name
```
Comment (hkantare) [inline modules/resource-group/README.md]: resource_group_name shd be name

File: modules/resource-group/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+## Inputs
+
+| Name                              | Description                                           | Type         | Default | Required |
+|-----------------------------------|-------------------------------------------------------|--------------|---------|----------|
+| name                              | Name of the resource group.                           | string       | n/a     | yes      |
+| tags                              | Tags associated with the resource group instance.     | list(string) | n/a     | no       |
```
Comment (hkantare) [inline modules/resource-group/README.md]: Tags associated with the resource group instance.  change to
Tags associated with the resource group.

File: modules/resource-group/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+## Note
+
+All optional fields are given value `null` in varaible.tf file. User can configure the same by overwriting with appropriate values.
+
+## Outputs
+
+| Name                              | Description                                                 |
+|-----------------------------------|-------------------------------------------------------------|
+| id                                | resource group id                                           |
```
Comment (hkantare) [inline modules/resource-group/README.md]: Follow common practise across description...Some are starting with small letters , some with captial letters

File: modules/resource-group/output.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+output "id" {
```
Comment (hkantare) [inline modules/resource-group/output.tf]: rename to resource_group_id

File: modules/resource-group/output.tf [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  description = "resource group id"
+  value       = var.provision ? ibm_resource_group.resource_group[0].id : data.ibm_resource_group.resource_group[0].id
+}
+
+output "default" {
```
Comment (hkantare) [inline modules/resource-group/output.tf]: Do we need this outputs both default or provision

Comment (umarali-nagoor) [reply modules/resource-group/output.tf]: Added as per cloud native toolkit resource group module ... if the GIS team needs it they can use them

https://github.com/cloud-native-toolkit/terraform-ibm-resource-group/blob/main/outputs.tf

**Pull request merged**